ethers-core/src/types/trace/geth.rs

```diff
@@ -90,11 +90,32 @@ impl From<NoopFrame> for GethTraceFrame {
 
 #[derive(Debug, PartialEq, Eq, Clone, Deserialize, Serialize)]
 #[serde(untagged)]
+pub enum GethTraceResult {
+    ResultKnown{result: GethTraceFrame},
+    ResultUnknown{result: Value},
+    DefaultKnown(GethTraceFrame),
+    DefaultUnknown(Value),
+}
+
+#[derive(Debug, PartialEq, Eq, Clone, Deserialize, Serialize)]
+#[serde(from = "GethTraceResult")]
+#[serde(untagged)]
 pub enum GethTrace {
     Known(GethTraceFrame),
     Unknown(Value),
 }
 
+impl From<GethTraceResult> for GethTrace {
+    fn from(value: GethTraceResult) -> Self {
+        match value {
+            GethTraceResult::DefaultKnown(t) => GethTrace::Known(t),
+            GethTraceResult::DefaultUnknown(v) => GethTrace::Unknown(v),
+            GethTraceResult::ResultKnown{result} => GethTrace::Known(result),
+            GethTraceResult::ResultUnknown{result} => GethTrace::Unknown(result),
+        }
+    }
+}
+
```

Comment:
This could probably be cleaned up.  It was done to convert a batch of traces to a `Vec<GethTrace>` since `debug_traceBlock` wraps each trace in a "result" key.

Example raw response:

```
{
    "jsonrpc": "2.0",
    "id": 1,
    "result": [
        {
            "result": {
                "from": "0x1b7aa44088a0ea95bdc65fef6e5071e946bf7d8f",
                "gas": "0x0",
                "gasUsed": "0x5208",
                "to": "0x7cb47db064be90f6ace1673ebfe5b3a102f369b8",
                "input": "0x",
                "value": "0xb1a2bc2ec50000",
                "type": "CALL"
            }
        },
        {
            "result": {
                "from": "0xe77e6c57b20b3de8c998b4552126feaeaf32b762",
                "gas": "0x2ee0",
...
```